examples/react-router/app/components/recipe_adder.jsx

```diff
@@ -0,0 +1,66 @@
+var React = require("react"),
+    Router = require("react-router"),
+    RouteHandler = Router.RouteHandler,
+    Link = Router.Link,
+    Navigation = Router.Navigation,
+    State = Router.State,
+    Fluxxor = require("../../../../");
+
+var RecipeForm = require("../forms/recipe_form.jsx");
+
+var RecipeAdder = React.createClass({
+  mixins: [
+    Fluxxor.FluxMixin(React),
+    State,
+    Navigation
+  ],
+
+  render: function() {
+    return this.renderWithLayout(
+      <div>
+        <form onSubmit={this.onSubmit}>
+          <RecipeForm ref="form" />
+          <input type="submit" value="Save" />
+        </form>
+      </div>
+    );
+  },
+
+  renderWithLayout: function(content) {
+    return (
+      <div>
+        {content}
+        <hr />
+        <Link to="home">Home</Link>
+        {" | "}<Link to="add-recipe">Add New Recipe</Link>
+      </div>
+    );
+  },
+
+  onSubmit: function(e) {
+    e.preventDefault();
+
+    var newRecipe = this.refs.form.getValue();
+    if (newRecipe) {
+      this.getFlux().actions.recipes.add(
+        newRecipe.name,
+        newRecipe.description,
+        newRecipe.ingredients,
+        newRecipe.directions,
+        function(recipe) {
```

Comment:
What I do in this situation is I have the store emit('create', {type: 'user', id: 2}) then at the top of the app I listen to that store and called router.transitionTo(payload.type, {id: payload.id});
Doing this outside of the store and in the core of the app lets you do things like loading ui for pending creates, then cancel redirection if the user tries to navigate away.